examples/cpp/game_mechanics/scene_management/src/main.cpp

```diff
@@ -0,0 +1,189 @@
+#include <scene_management.h>
+#include <iostream>
+
+// This example shows one possible way to implement scene management
+// using pipelines.
+
+// Scene relationships/tags
+struct ActiveScene { }; // Represents the current scene
+struct SceneRoot { }; // Parent for all entities unique to the scene
+
+// Scenes
+using Pipeline = flecs::entity;
+struct MenuScene { Pipeline pip; };
+struct GameScene { Pipeline pip; };
+
+// Components for Example
+struct Position { float x, y; };
+struct Button { std::string text; };
+struct Character { bool alive; };
+struct Health { int amount; };
+
+// Removes all entities who are children of
+// the current scene root and returns it.
+// (NOTE: should use defer_begin() / defer_end())
+flecs::entity reset_scene(flecs::world& ecs) {
+    auto scene = ecs.component<SceneRoot>();
```

Comment:
You could do this here, which is a bit more efficient:

```cpp
ecs.delete_with(flecs::ChildOf, ecs.entity<SceneRoot>());
```

Reply:
Thanks, fixed with 1217d63